test/e2e/upgrade_test.go

```diff
@@ -78,6 +92,30 @@ func testUpgradeVersion(t *testing.T) {
 
 }
 
+// deployByohWorkload deploys a workload in a BYOH node, using BYOH-label affinity as NodeSelector
+func (tc *testContext) deployByohWorkload() (*appsv1.Deployment, error) {
```

Comment:
Why does this need to be a separate function from the general `deployWindowsWorkloadAndTester()`?
 https://github.com/openshift/windows-machine-config-operator/blob/7c4ed1e3950757d077ba7482f39c4af68ff25429/test/e2e/upgrade_test.go#L157

That function could be modified to take an affinity.

Reply:
Sure, we can move in that direction. The only show stopper is that the[ deployment get deleted ](https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/upgrade_test.go#L163)before the upgrade process is triggered

Reply:
@jrvaldes to reuse logic you could  modify existing function deployWindowsWorkloadAndTester() to add the affinity parameter and call it in the upgrade test suite wherever required . One way to deal with the issue you are facing could be to can move the creation of deployment outside of the function and use that name to delete the deployment at the end of the suite.